apriltag_quad_thresh.c

```diff
@@ -714,8 +714,16 @@ static inline void ptsort(struct pt *pts, int sz)
 #undef MAYBE_SWAP
 
     // a merge sort with temp storage.
-
-    struct pt *tmp = malloc(sizeof(struct pt) * sz);
+    // Use stack allocation for small arrays to avoid malloc overhead
+    #define STACK_BUFFER_SIZE 256
+    struct pt stack_buffer[STACK_BUFFER_SIZE];
+    struct pt *tmp;
+    bool use_heap = sz > STACK_BUFFER_SIZE;
```

Comment:
Make this a `const bool use_heap` as it is not changing afterwards. Squash this change into the same commit.